segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -144,8 +144,12 @@ public void close() {
     @Override
     public CompletableFuture<Void> deleteSegment(@NonNull String segmentName, boolean mustBeEmpty, Duration timeout) {
         if (mustBeEmpty) {
-            // TODO https://github.com/pravega/pravega/issues/2956
-            throw new UnsupportedOperationException("deleteSegment(mustBeEmpty==true)");
+            TimeoutTimer timer = new TimeoutTimer(timeout);
+            return this.segmentContainer
+                    .forSegment(segmentName, timer.getRemaining())
+                    .thenComposeAsync(segment -> this.keyIndex.executeIfEmpty(segment,
```

Comment:
Say I'm  trying to delete a table segment. When the call gets here, it fails because it is not empty. Would the user of the table segment get an iterator so that it can determine what keys are still in there? I'm trying to understand what the next steps are in the case an attempt for deleting the table segment fails.

Reply:
The user would get an exception indicating it's not empty.

Then the user can do whatever they want. If they want to call `keyIterator`, they can definitely do that.